gtsfm/graph_partitioner/binary_tree_partition.py

```diff
@@ -0,0 +1,129 @@
+from typing import Dict, List, Tuple
+
+import gtsam
+import networkx as nx
+from gtsam import SymbolicFactorGraph
+
+import gtsfm.utils.logger as logger_utils
+from gtsfm.graph_partitioner.graph_partitioner_base import GraphPartitionerBase
+
+logger = logger_utils.get_logger()
+
+
+class BinaryTreeNode:
+    def __init__(self, keys: List[int], depth: int):
+        self.keys = keys  # Only at leaves
+        self.left = None
+        self.right = None
+        self.depth = depth
+
+    def is_leaf(self) -> bool:
+        return self.left is None and self.right is None
+
+
+class BinaryTreePartition(GraphPartitionerBase):
+    def __init__(self, max_depth: int = 2):
+        super().__init__(process_name="BinaryTreePartition")
+        self.max_depth = max_depth
+
+    def partition_image_pairs(self, image_pairs: List[Tuple[int, int]]) -> List[List[Tuple[int, int]]]:
+        if not image_pairs:
+            logger.warning("No image pairs provided for partitioning.")
+            return []
+
+        symbol_graph, _, nx_graph = self._build_graphs(image_pairs)
+        ordering = gtsam.Ordering.MetisSymbolicFactorGraph(symbol_graph)
+        self.root = self._build_binary_partition(ordering)
+
+        num_leaves = 2**self.max_depth
+        partition_details = [{} for _ in range(num_leaves)]
+        image_pair_partitions = [[] for _ in range(num_leaves)]
+
+        self._collect_leaf_partitions(self.root, nx_graph, partition_details, leaf_idx=[0])
```

Comment:
functionality again, would be nice to just create partition_details in that method and return it.